ivy/src/main/scala/sbt/librarymanagement/ivy/Credentials.scala

```diff
@@ -83,4 +83,5 @@ final class DirectCredentials(
     val host: String,
     val userName: String,
     val passwd: String
+  override def toString = s"DirectCredentials('$realm', '$host', '$userName', ****)"
```

Comment:
`val passwd: String` is in the param list, so you need `) extends Credentials { ... }` block.

Reply:
Doh sorry :/